src/api/app/views/webui/users/tokens/edit.html.haml

```diff
@@ -8,20 +8,73 @@
+                    %span.input-group-text
+                      %i.fas.fa-clipboard
+              %hr
+
+          .form-row
             .col-12.col-md-10.col-lg-9
               .form-group
                 = f.label(:id, 'Id:')
                 = @token.id
               .form-group
                 = f.label(:type, 'Operation:')
                 = @token.class.token_name.capitalize
-              .form-group
-                = f.label(:scm_token) do
-                  SCM token:
-                  %abbr.text-danger{ title: 'required' } *
-                .input-group
-                  = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your new SCM token', required: true)
+              .form-group#regenerate-token-group
+                = link_to('Regenerate Token', '#', title: 'Regenerate Token', class: 'btn btn-outline-danger float-right ml-3 px-4',
+                  data: { toggle: 'modal', target: '#regenerate-token-modal' })
+                If your token has been lost, forgotten or compromised, you can regenerate it.
+                Don't forget to replace the string in any scripts or applications using this token.
+              - if @token.type == 'Token::Workflow'
+                .form-group#scm-token-group
+                  = f.label(:scm_token) do
+                    SCM token:
+                    %abbr.text-danger{ title: 'required' } *
+                  .input-group
+                    = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your new SCM token', required: true)
           .actions
             = link_to('Cancel', tokens_path, title: 'Cancel', class: 'btn btn-outline-secondary px-4 mr-3 mt-3 mt-sm-0')
-            = f.submit('Update', class: 'btn btn-primary px-4 mt-3 mt-sm-0')
+            - if @token.type == 'Token::Workflow'
+              = f.submit('Update', class: 'btn btn-primary px-4 mt-3 mt-sm-0')
+
+.modal.fade#regenerate-token-modal{ tabindex: -1, role: 'dialog', aria: { labelledby: 'regenerate-token-modal-label', hidden: true } }
+  .modal-dialog.modal-dialog-centered{ role: 'document' }
+    .modal-content
+      .modal-header
+        %h5.modal-title#regenerate-token-modal-label Do you really want to regenerate this token string?
+      .modal-body
+        %p
+          Please confirm that you want to regenerate the token string of the token with id
+          = surround "'" do
+            = @token.id
+        = form_for(@token, as: :token, url: token_path, method: :put, remote: true) do |f|
+          .modal-footer
+            %a.btn.btn-sm.btn-outline-secondary.px-4{ data: { dismiss: 'modal' } }
+              Cancel
+            -# %input.btn.btn-sm.btn-danger.px-4{ type: 'submit', name: 'commit', value: 'Regenerate', data: { disable: { with: 'Regenerate' } } }
+            = f.submit('Regenerate', class: 'btn btn-sm btn-danger px-4', data: { disable: { with: 'Regenerate' } })
+
+:javascript
```

Comment:
The JS code in this file is the same as what we have in `src/api/app/views/webui/users/tokens/new.html.haml`:

https://github.com/openSUSE/open-build-service/blob/b36fc16e3f985db371d04f5b7dbb469e846ecdcb/src/api/app/views/webui/users/tokens/new.html.haml#L54-L58

https://github.com/openSUSE/open-build-service/blob/b36fc16e3f985db371d04f5b7dbb469e846ecdcb/src/api/app/views/webui/users/tokens/new.html.haml#L79-L87

Any plans to remove this duplication?

Reply:
> Any plans to remove this duplication?
Yes, I was on it.

> This input with the copy-to-clipboard feature could be a view component.
Ok, I will create a view component.

Reply:
@eduardoj and I talked. We created the view component `CopyToClipboardInputComponent`.